buildkite/resource_pipeline_test.go

```diff
@@ -105,7 +105,54 @@ func TestAccPipeline_add_remove_withcluster(t *testing.T) {
 					testAccCheckPipelineRemoteValues(&resourcePipeline, "Test Pipeline foo"),
 					// Confirm the pipeline has the correct values in terraform state
 					resource.TestCheckResourceAttr("buildkite_pipeline.foobar", "name", "Test Pipeline foo"),
-					resource.TestCheckResourceAttr("buildkite_pipeline.foobar", "cluster_id", ""),
+					resource.TestCheckNoResourceAttr("buildkite_pipeline.foobar", "cluster_id"),
+					resource.TestCheckResourceAttr("buildkite_pipeline.foobar", "allow_rebuilds", "true"),
+				),
+			},
+		},
+	})
+}
+
+// TestAccPipeline_add_remove_withcluster_old_version tests that the provider doesn't get stuffed up when coming from an
+// older version to this new version. the first step runs against the published v0.21.1 version and the second step runs
+// against the current provider. the configuration is the same between each one so there should be no state/plan change
+func TestAccPipeline_add_remove_withcluster_old_version(t *testing.T) {
```

Comment:
I'd like specific and thorough review of these additional tests. Their aim is ensure that the state remains the same during the upgrade path of users. The first step runs against the published version of the provider. The second step uses the same config but runs against the framework version of the resource

Reply:
Sound - given `0.21.1` containing SDKv2 implementation - ensuring the later versions as the provider stays consistent and retains the functionality upon what we expect to happen between old/new